packages/vite/src/node/plugins/resolve.ts

```diff
@@ -155,14 +162,38 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
         return optimizedPath
       }
 
+      const ensureVersionQuery = (resolved: string): string => {
+        if (
+          !options.isBuild &&
+          depsOptimizer &&
+          !(
+            resolved === normalizedClientEntry ||
+            resolved === normalizedEnvEntry
+          )
+        ) {
+          // Ensure that direct imports of node_modules have the same version query
+          // as if they would have been imported through a bare import
+          // Use the original id to do the check as the resolved id may be the real
+          // file path after symlinks resolution
+          const isNodeModule = !!normalizePath(id).match(nodeModulesInPathRE)
```

Comment:
It looks like most of the codebase simply checks by `includes('node_modules')`. I think we can probably do the same for consistency even if there might be false positives. But I don't feel strongly otherwise if we don't change this too.

Reply:
We have a good mix now, other places tests for `includes('/node_modules/')`. Let's merge this as I think we should be more strict on this check, and we could update places only checking for `'node_modules'` in future PRs.